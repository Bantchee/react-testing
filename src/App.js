import React, { useState } from "react";

const App = () => {
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    (heading === "Magnificent Monkeys") ?
      setHeading("Radical Rhinos") :
      setHeading("Magnificent Monkeys");
  };

  return (
    <div>
      <button
        type="button"
        onClick={clickHandler}
      >
        Click Me!
      </button>
      <h1>{heading}</h1>
    </div>
  )
}

export default App;