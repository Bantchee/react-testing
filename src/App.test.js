import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App compnent", () => {
  test("renders Magnificent Monkeys", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot()
  });

  test("renders Radical Rhinos after button click", () => {
    render(<App />);
    const button = screen.getByRole("button", { name: "Click Me!" });

    userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/Radical Rhinos/i);
  });
});
