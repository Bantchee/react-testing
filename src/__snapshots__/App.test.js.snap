// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App compnent renders Magnificent Monkeys 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      Click Me!
    </button>
    <h1>
      Magnificent Monkeys
    </h1>
  </div>
</div>
`;
